### DISCUSSION TITLE
Package update after adding 3.0.0-dev-preview

### DISCUSSION THREAD

**Discussion by lobunto** [sanity-io/sanity#3556] [Sanity Studio Support] [answered]
Hello! I am working on upgrading a project from V2 to V3. After adding `"sanity": "3.0.0-dev-preview"` to `package.json`, I get some errors, such as:

- `Could not resolve "part:@sanity/base/datastore/document": node_modules/@sanity/react-hooks/lib/useEditState.js:8:47`
- `Could not resolve "part:@sanity/base/schema":   node_modules/@sanity/initial-value-templates/lib/parts/Schema.js:14:54`
- `Could not resolve "part:@sanity/base/user": node_modules/@sanity/base/lib/datastores/user/hooks.js:9:43`

I wonder if any of the `@sanity` packages I have installed need a newer version. Or are some already included in the "new" `sanity`-package? 

I run `npx sanity start` in the terminal to start the project.
Here are the sanity packages that I've currently installed:

{
  ...
    "@sanity/base": "2.30.0",
    "@sanity/cli": "2.30.2",
    "@sanity/core": "2.30.0",
    "@sanity/default-layout": "2.30.0",
    "@sanity/default-login": "2.30.0",
    "@sanity/desk-tool": "2.30.0",
    "@sanity/eslint-config-studio": "2.0.0",
    "@sanity/image-url": "1.0.1",
    "@sanity/vision": "2.30.0",
    "sanity": "3.0.0-dev-preview.8",
  ...
}

**Comment by dpfavand** [comment, marked as answer]
👋 Great question.

As you guessed, a lot of those packages are now included in the `sanity` package.

My recommendation at this time would be to init a new v3 project in a new directory alongside your v2 studio. Then copy over the configurations you need and install the v3 plugins you want. That should be pretty straightforward overall and simpler.

Let us know if that gets you unblocked?

**Reply by lobunto** [reply]
@dpfavand 👋🏻 I created a new project with v3, then copied and installed each package, and then ran the project for each package installed. After some try and fail, it seems that just removing `@sanity/core` did the trick. I also had to fix all the imports that were using `@sanity/*` and instead import from the `sanity` package.

So, it seems to work fine now. Thanks for the suggestion!

**Reply by dpfavand** [reply]
Glad you got it working. From the list above, I believe you won't need any of these ones, if you have them installed I don't believe they are being used, as they've been rolled into the `sanity` package:
```
@sanity/base": "2.30.0",
"@sanity/cli": "2.30.2",
"@sanity/core": "2.30.0",
"@sanity/default-layout": "2.30.0",
"@sanity/default-login": "2.30.0",
"@sanity/desk-tool": "2.30.0",
```
